File: 3R/10400_The3n+1problem.cpp
#include <iostream>

using namespace std;

int count_3n(long n){
	long count = 1;

	while(n != 1){
		if(n % 2 == 0) n/=2;
		else n = 3*n + 1;

		count++;
	}

	return count;
}

int main(){
	long a, b;
	

	while(cin >> a >> b){
		int max_count = 0;

		cout << a << " " << b << " ";

		if(a > b){
			long tmp = a;
			a = b;
			b = tmp;
		}

		for(long i = a; i <= b; i++){
			max_count = max(count_3n(i), max_count);
		}

		cout << max_count << endl;
	}
}

File: 3R/10400_The3n+1problem.java
//10400:3n+1

import java.util.*;

class main{

	static int calc_3n(long n){
		int count = 1;
		
		while(n!=1){
			if(n%2==0) n/=2;
			else n = 3 * n + 1;

			count ++;
		}

		return count;
	}

	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
		while(sc.hasNext()){
			long a = sc.nextLong();
			long b = sc.nextLong();

			System.out.print(a + " " + b + " ");
			int max = 0;

			if(a>b){
				long tmp = a;
				a = b;
				b = tmp;
			}

			for(long i=a; i<=b; i++){
				max = Math.max(calc_3n(i), max);
			}

			System.out.println(max);

		}
	}
}